sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py

```diff
@@ -458,12 +462,13 @@ def receive_deferred_messages(self, sequence_numbers):
         messages = self._mgmt_request_response_with_retry(
             REQUEST_RESPONSE_RECEIVE_BY_SEQUENCE_NUMBER,
             message,
-            handler
+            handler,
+            timeout=timeout
         )
         return messages
 
-    def peek_messages(self, max_message_count=1, sequence_number=None):
-        # type: (int, Optional[int]) -> List[PeekedMessage]
+    def peek_messages(self, max_message_count=1, sequence_number=None, **kwargs):
```

Comment:
Do not take this as a "do this" comment, take this as me grumbling out loud _in case_ it changes your outlook, but if you already contemplated this then just ignore my curmudgeoning:

Never loved purely kwarging things that there wasn't a forcing function for (e.g. *locktokens prior).  It makes IDE hinting harder, hides the option deeper (which may partly be the goal?)  In any case, you know my stance on this, but just taking this opportunity to make my normal handwaving :P  @annatisch (this comment is as much for you TBH, and same statement, feel free to wave this off as 'Kieran and his weird philosophies', figured I'd just throw this out as a datapoint)

Reply:
`timeout` should definitely be in kwargs - it's a pretty advanced/narrow usecase, and in most Python standard libraries this level of configuration would be relegated to kwargs.
In addition, our guidelines are pretty clear on it:
https://azure.github.io/azure-sdk/python_introduction.html#method-signatures

Reply:
following the guideline, timeout moved into kwargs and along with sequence_number.